client/dive-common/components/Viewer.vue

```diff
@@ -338,15 +344,25 @@ export default defineComponent({
         videoUrl.value = meta.videoUrl;
         datasetType.value = meta.type as DatasetType; // TODO: support for multiCam will remove this
       }),
-      loadDetections(props.id).then((tracks) => {
-        Object.values(tracks).forEach(
-          (trackData) => insertTrack(Track.fromJSON(trackData), { imported: true }),
-        );
+      loadDetections(props.id).then(async (trackData) => {
+        const tracks = Object.values(trackData);
+        progress.total = tracks.length;
+        for (let i = 0; i < tracks.length; i += 1) {
+          if (i % 4000 === 0) {
```

Comment:
Do we want to add a  ` && i >= 4000` to this because if not for small datasets it does this quick transition between loading and 0% then everything is loaded.

Reply:
Also also sure.

Reply:
Fixed this in a different way in 15b52cf